compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -4313,6 +4313,11 @@ private void defineInvokableSymbol(BLangInvokableNode invokableNode, BInvokableS
         invokableNode.symbol = funcSymbol;
         defineSymbol(invokableNode.name.pos, funcSymbol);
         invokableEnv.scope = funcSymbol.scope;
+        SymbolEnv enclEnv = invokableEnv.enclEnv;
+        NodeKind enclEnvNodeKind = enclEnv.node.getKind();
+        if (enclEnvNodeKind == NodeKind.CLASS_DEFN || enclEnvNodeKind == NodeKind.OBJECT_TYPE) {
+            invokableEnv.enclEnv = enclEnv.enclEnv;
+        }
```

Comment:
IMO After creating a `SymbolEnv` changing `enclEnv` of that `env` is not correct. 
by doing `invokableEnv.enclEnv = enclEnv.enclEnv;` We set  `packageEnv` as `enclEnv` of `invokableEnv`

As I understand the parameters of methods(in class) cannot see the fields of that class. So when creating `invokableEnv` for methods(in class or object) we need to use class env without fields in the scope.

Reply:
Changed the login, please have a look